### PR TITLE
Push forced with lease when Commit & Push action is performed with Amend option checked.

### LINKED ISSUE #4296: Commit & Push (forced with lease) when Amend is checked
**Do you want to request a *feature* or report a *bug*?**

feature

**What is the expected behavior?**

When amending a commit the Commit & Push button is of no use because we already know that server will reject such a commit. I propose to add an advanced option to perform Commit & Push Forced with lease when users amend changes with the Commit & Push button.

### PR DESCRIPTION
**Pull request by jbialobr**
Closes #4296.

The Commit & Push button is rendered in red when it will push forced with lease.

### CODE REVIEW

File: GitUI/CommandsDialogs/FormCommit.cs [hunk 1]
```diff
@@ -2729,6 +2730,15 @@ private void Amend_CheckedChanged(object sender, EventArgs e)
...
                 Message.Text = Module.GetPreviousCommitMessages(1).FirstOrDefault()?.Trim();
             }
+            UpdateCommitAndPushButton();
+        }
+
+        private void UpdateCommitAndPushButton()
+        {
+            if (AppSettings.CommitAndPushForcedWhenAmend)
+            {
+                CommitAndPush.BackColor = Amend.Checked ? Color.Salmon : SystemColors.ButtonFace;
```
Comment (RussKie) [inline GitUI/CommandsDialogs/FormCommit.cs]: we need to ensure this will look good on Windows with dark/custom theme.
Maybe it will need to be configurable (can be in a separate PR in that case)

**Comment by vbjay** [comment]
Good point.

File: GitUI/CommandsDialogs/FormCommit.cs (hunk 1 above)
Comment (jbialobr) [reply GitUI/CommandsDialogs/FormCommit.cs]: Now it looks like that:
gitext
Is it good?

Comment (RussKie) [reply GitUI/CommandsDialogs/FormCommit.cs]: I suppose it is ok, commit looks horrible though, heh.

**Comment by RussKie** [comment]
I'll let translators have another round of translations before the final release.
There are a few translations under 95%
